xDripG5/Transmitter.swift

```diff
@@ -50,6 +51,8 @@ public final class Transmitter: BluetoothManagerDelegate {
 
     private var delegateQueue = DispatchQueue(label: "com.loudnate.xDripG5.delegateQueue", qos: .utility)
 
+    private var messageQueue = MessageQueue()
```

Comment:
The biggest concern I have here is that this doesn't support persistence, and that's important if the app were to be closed before being awoken when the transmitter appears next.

I don't think this framework should manage persistence, so I think the better approach is to ask the TransmitterDelegate for its pending commands, and to publicly expose a Command enum.

It's also important to allow re-queing of failed commands, and this should be up to the delegate to determine.

```swift
enum Command: RawRepresentable {
    typealias RawValue = Data

    case startSensor(at: Date())
    case stopSensor(at: Date())
    case calibrateSensor(to: HKQuantity())

    // Provide serialization for clients to easily persist and restore
}

public protocol TransmitterDelegate: class {
    func dequeuePendingCommand(for transmitter: Transmitter) -> Command?

    func transmitter(_ transmitter: Transmitter, didFail command: Command, with error: Error)

    func transmitter(_ transmitter: Transmitter, didComplete command: Command)
}
```

Reply:
Agreed. I did wonder how to manage persistence - this is a better solution.